### PR TITLE
Update PR_template

### PR DESCRIPTION
**Pull request by monsij**
Addresses #1208 
@ninamiolane , @chicken-biryani let me know if I am missing something !

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
-Please make sure that you that you that your pull request aligns with our checklist-- https://github.com/chicken-biryani/geomstats/blob/master/docs/contributing.rst#pull-request-checklist
 -->
 
+## Checklist
+
+- [ ] My pull request has a suggestive title.
+- [ ] If neccessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
+- [ ] I have added apropriate unit tests.
+- [ ] I have made sure the code passes all unit tests.
+- [ ] My PR follows PEP8 guidelines (run '`flake8`)
```
Comment (chicken-biryani) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: The link to Pep8

Comment (chicken-biryani) [reply .github/PULL_REQUEST_TEMPLATE.md:11]: Comment about the commands to run can be added?
"$ pip3 install -r dev-requirements.txt
Then you can run the following two commands:

$ flake8 --ignore=D,W503,W504 geomstats examples tests
$ flake8 geomstats/geometry geomstats/learning"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
+- [ ] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
+- [ ] My code is properly documented and I made sure the documentation renders properly.
```
Comment (chicken-biryani) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: Link to the documentation style.

**Review by chicken-biryani — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
-
-Please make sure that you that you that your pull request aligns with our checklist-- https://github.com/chicken-biryani/geomstats/blob/master/docs/contributing.rst#pull-request-checklist
 -->
 
+## Checklist
+
+- [ ] My pull request has a suggestive title.
+- [ ] If neccessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
+- [ ] I have added apropriate unit tests.
+- [ ] I have made sure the code passes all unit tests.
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: Great comments. Additionally, you can explain how to change the backend `export GEOMSTATS_BACKEND=pytorch` to test in pytorch, for example.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,17 +1,26 @@
 <!--
 Thank you for opening this pull request!
-
-Please make sure that you that you that your pull request aligns with our checklist-- https://github.com/chicken-biryani/geomstats/blob/master/docs/contributing.rst#pull-request-checklist
 -->
 
+## Checklist
+
+- [ ] My pull request has a suggestive title.
```
Comment (ninamiolane) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: "clear and explanatory" title

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,17 +1,26 @@
...
+
 ## Description 
 
-<!-- Include a description of your pull request. -->
+<!-- Include a description of your pull request. If relevant, feel free to use this space to talk about time and space complexity as well scalability of your code-->
 
 ## Issue
 
-<!-- Tell us which issue does this PR fix -->
+<!-- Tell us which issue does this PR fix . Why is this feature implementation/fix is important in practice ?-->
```
Comment (ninamiolane) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: typo: remove double "is"

**Review by ninamiolane — COMMENTED** [review]
Thanks Monsij! A few comments

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (monsij) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added commented sections below checklist to make it concise !

**Comment by monsij** [comment]
@ninamiolane @chicken-biryani changes made !

**Review by ninamiolane — APPROVED** [review]
LGTM, many thanks!

**Pull request merged**